apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -2563,6 +2563,10 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
 			.toArray();
 	}
 
+	countLivechatRoomsWithDepartment(): Promise<number> {
+		return this.col.countDocuments({ departmentId: { $exists: true }, t: 'l' });
```

Comment:
I'd recommend to drop the `t: 'l'` filter so we can make good use of the sparse index on `departmentId`.. and we're sure whenever a room has a `departmentId` it will be `t: 'l'`..
```suggestion
		return this.col.countDocuments({ departmentId: { $exists: true } });
```

JFYI I have tested this in a collection with 1.7mi rooms and removing the `t: 'l'` from the filter reduced the query time from 34s to 800ms